week2/homework/src/actions.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const fs = require('fs');
+
+const ENCODING = 'utf8';
+const PATH = `${__dirname}/todoList.json`;
+
+const actions = {
+  list: () => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      console.log(`\nTodo list:\n${todoList}`);
+    });
+  },
+  add: todoItem => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      todoList.push(todoItem);
+      fs.writeFile(PATH, JSON.stringify(todoList), err => {
+        if (err) throw err;
+      });
+      console.log(`'${todoItem}' has been added to the Todo list.`);
+    });
+  },
+  remove: index => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      if (index > 0 && index <= todoList.length) {
+        todoList.splice(index - 1, 1);
+        console.log('Todo item has been removed');
```

Comment:
This message should be moved to the callback, so you don't report the success early.

Reply:
Thanks... done 👍